website/docs/en/api/javascript-api/instance.mdx

```diff
@@ -362,6 +360,27 @@ const server = await rsbuild.createDevServer();
 await server.listen();
 ```
 
+### connectSocket
+
+Rsbuild has a builtin Socket server to support HMR:
+
+1. When a user accesses a page through browser, a Web Socket connection request is automatically initiated to the server.
+2. After the Rsbuild dev server detects the connection request, it instructs the builtin Socket server to process it.
+3. After the browser successfully establishes a connection with the Rsbuild Socket server, real-time communication is possible.
+4. The Rsbuild Socket server notifies the browser after each recompilation is complete. The browser then sends a `hot-update.(js|json)` request to the dev server to load the new compiled module.
+
+When you use custom server, you may encounter HMR connection error problems. This is because the custom server does not forward Web Socket connection requests to Rsbuild's Socket server.
```

Comment:
Is it a `Socket server` or a `Socket middleware`? Rsbuild does not start server when using custom server.

Reply:
how about `Socket handler`?Users will misunderstand socket middleware and other middlewares (why `app.use(rsbuildServer.middlewares)` is useless)

Reply:
LGTM